photon-api/src/main/scala/com/linkedin/photon/ml/model/RandomEffectModel.scala

```diff
@@ -126,56 +117,47 @@ class RandomEffectModel(
    *
    * @return The Spark context
    */
-  override protected[ml] def sparkContext: SparkContext = modelsRDD.sparkContext
+  override protected[ml] def sparkContext: SparkContext = SparkSession.builder.getOrCreate.sparkContext
 
-  /**
-   * Assign a given name to [[modelsRDD]].
-   *
-   * @note Not used to reference models in the logic of photon-ml, only used for logging currently.
-   * @param name The parent name for all [[RDD]]s in this class
-   * @return This object with the name of [[modelsRDD]] assigned
-   */
   override protected[ml] def setName(name: String): RandomEffectModel = {
 
-    modelsRDD.setName(name)
-
     this
   }
 
   /**
-   * Set the storage level of [[modelsRDD]], and persist their values across the cluster the first time they are
+   * Set the storage level of [[models]], and persist their values across the cluster the first time they are
    * computed.
    *
    * @param storageLevel The storage level
-   * @return This object with the storage level of [[modelsRDD]] set
+   * @return This object with the storage level of [[models]] set
```

Comment:
All of the `RDDLike` functions should be deleted and so should the `RDDLike` class.